applications/tari_merge_mining_proxy/src/proxy.rs

```diff
@@ -296,96 +305,118 @@ impl InnerService {
+                        hex::encode(&hash)
+                    );
+                    continue;
                 },
             };
 
-            debug!(target: LOG_TARGET, "Located Tari Hash: {:?}", hash);
-
-            if let Some(mut block_data) = self.block_templates.get(&hash).await {
-                let monero_data = helpers::construct_monero_data(monero_block, block_data.clone().monero_seed)?;
-
-                let pow_data = bincode::serialize(&monero_data)?;
-                let h = block_data.tari_block.header.as_mut().unwrap();
-                let height = h.height;
-                let p = h.pow.as_mut().unwrap();
-                p.pow_data = pow_data;
-
-                let mut base_node_client = self.connect_grpc_client().await?;
-                let mut block_templates_clone = self.block_templates.clone();
-                let block_data_clone = block_data.clone();
-                handle.spawn(async move {
-                    let start = std::time::Instant::now();
-                    match base_node_client
-                        .submit_block(block_data_clone.tari_block)
-                        .await
-                        .map_err(|status| MmProxyError::GrpcRequestError {
-                            status,
-                            details: "failed to submit block".to_string(),
-                        }) {
-                        Ok(..) => {
-                            debug!(
-                                target: LOG_TARGET,
-                                "Submitted block #{} to Tari node in {:.0?} (SubmitBlock)",
-                                height,
-                                start.elapsed()
-                            );
-                            block_templates_clone.remove(&hash).await;
-                        },
-                        _ => debug!(
-                            target: LOG_TARGET,
-                            "Problem submitting block #{} to Tari node, responded in  {:.0?} (SubmitBlock)",
-                            height,
-                            start.elapsed()
-                        ),
+            let monero_data = helpers::construct_monero_data(monero_block, block_data.monero_seed.clone())?;
+
+            let header_mut = block_data.tari_block.header.as_mut().unwrap();
+            let height = header_mut.height;
+            header_mut.pow.as_mut().unwrap().pow_data = bincode::serialize(&monero_data)?;
+
+            let mut base_node_client = self.connect_grpc_client().await?;
+            let start = Instant::now();
+            match base_node_client.submit_block(block_data.tari_block).await {
+                Ok(_) => {
+                    json_resp = json_rpc_success_response(
+                        json_resp["id"].as_i64(),
+                        json!({
+                            "status": json_resp["result"]["status"],
+                            "tari_status": "OK",
```

Comment:
"merge_mining": "Ok"

Has to be generic enough to be used by any coin.

Reply:
I think the goal here is to have an API that can be used by a supporting consumer that is using the MM proxy, not to create a generic solution to merge mining with Monero in general (although the learnings from this could be used to create that spec). 

Perhaps something like

```javascript
{
  "id": 123,
  ... etc
  "result": {
    "status": "OK",
    "_merge_mining": { // underscore because non-standard, non-stable 
      "statuses": {"xtr": "OK"}
   }
  }
}
```

Reply:
With the exception of the tari specific grpc calls we make witin the proxy, the solution is pretty generic already.

I agree with your above suggestion, that way statuses can just be displayed off an iterator. Nice.

Reply:
I've decided not to include any extension data in the call since it probably isn't needed and it keeps things simple. Especially since the monerod api allows multiple blocks to be submitted in one call. 

The only change made to this call is to return a success result if Tari and/or Monero succeed.